kobodl/kobo.py

```diff
@@ -447,53 +467,23 @@ def LoadInitializationSettings(self) -> None:
             print(response.reason, response.text)
             raise err
 
-    def Login(self, email: str, password: str, captcha: str) -> None:
-        (
-            signInUrl,
-            workflowId,
-            requestVerificationToken,
-        ) = self.__GetExtraLoginParameters()
-
-        postData = {
-            "LogInModel.WorkflowId": workflowId,
-            "LogInModel.Provider": Kobo.Affiliate,
-            "ReturnUrl": "",
-            "__RequestVerificationToken": requestVerificationToken,
-            "LogInModel.UserName": email,
-            "LogInModel.Password": password,
-            "g-recaptcha-response": captcha,
-            "h-captcha-response": captcha,
-        }
-
-        response = self.Session.post(signInUrl, data=postData)
-        debug_data("Login", response.text)
-        response.raise_for_status()
-        htmlResponse = response.text
-
-        match = re.search(r"'(kobo://UserAuthenticated\?[^']+)';", htmlResponse)
-        if match is None:
-            soup = BeautifulSoup(htmlResponse, 'html.parser')
-            errors = soup.find(class_='validation-summary-errors') or soup.find(
-                class_='field-validation-error'
-            )
-            if errors:
-                raise KoboException('Login Failure! ' + errors.text)
-            else:
-                with open('loginpage_error.html', 'w') as loginpagefile:
-                    loginpagefile.write(htmlResponse)
-                raise KoboException(
-                    "Authenticated user URL can't be found. The page format might have changed!\n\n"
-                    "The bad page has been written to file 'loginpage_error.html'.  \n"
-                    "You should open an issue on GitHub and attach this file for help: https://github.com/subdavis/kobo-book-downloader/issues\n"
-                    "Please be sure to remove any personally identifying information from the file."
-                )
-
-        url = match.group(1)
-        parsed = urllib.parse.urlparse(url)
-        parsedQueries = urllib.parse.parse_qs(parsed.query)
-        self.user.UserId = parsedQueries["userId"][
-            0
-        ]  # We don't call self.Settings.Save here, AuthenticateDevice will do that if it succeeds.
-        userKey = parsedQueries["userKey"][0]
-
+    def Login( self ) -> None:
+        activationCheckUrl, activationCode = self.__ActivateOnWeb()
+
+        print("")
+        print("kobo-book-downloader uses the same web-based activation method to log in as the")
+        print("Kobo e-readers. You will have to open the link below in your browser and enter")
+        print("the code. You might need to login if kobo.com asks you to.")
+        print("")
+        print(f"Open https://www.kobo.com/activate and enter {activationCode}.")
```

Comment:
I suggest removing the `.` from the end. Removing would make it easier to select the PIN without also grabbing the dot.